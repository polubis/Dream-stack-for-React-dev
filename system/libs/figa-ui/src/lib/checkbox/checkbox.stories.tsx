import type { Story, Meta } from '@storybook/react';

import { Checkbox } from './checkbox';
import { Box } from '../box';
import { Font } from '../font';

export default {
  component: Checkbox,
  title: 'Checkbox',
} as Meta;

const Template: Story = () => {
  return (
    <Box padding={[300, 300, 300, 300]}>
      <Box spacing={[200, 200]}>
        <Font variant="h5">Possible use cases</Font>
        <Box orientation="row" spacing={[300, 300, 300]}>
          <Checkbox label="Unchecked" />
          <Checkbox label="Checked" checked />
          <Checkbox label="Disabled and unchecked" disabled />
          <Checkbox label="Disabled and checked" disabled checked />
        </Box>
        <Box maxWidth="max-content" spacing={[300]}>
          <Checkbox label="Reversed" reversed />
          <Checkbox />
        </Box>
      </Box>
    </Box>
  );
};

export const Default = Template.bind({});
Default.args = {};
