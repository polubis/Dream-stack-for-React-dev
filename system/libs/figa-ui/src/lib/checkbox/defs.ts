import type { ReactNode, DetailedHTMLProps, InputHTMLAttributes } from 'react';

type CheckboxHTMLElementProps = Omit<
  DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>,
  'type' | 'value' | 'children'
>;

interface CheckboxProps extends CheckboxHTMLElementProps {
  reversed?: boolean;
  label?: ReactNode;
}

export type { CheckboxProps };
