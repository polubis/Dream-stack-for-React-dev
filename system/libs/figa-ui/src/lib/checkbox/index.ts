export * from './checkbox';
export * from './defs';
