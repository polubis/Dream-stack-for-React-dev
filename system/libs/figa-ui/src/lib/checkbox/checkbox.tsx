import styled from 'styled-components';

import { CheckboxProps } from './defs';

const StyledCheckbox = styled.input.attrs({ type: 'checkbox' })`
  width: 300px;
  height: 300px;
`;

const Checkbox: React.FC<CheckboxProps> = () => {
  return <StyledCheckbox />;
};

export { Checkbox };
