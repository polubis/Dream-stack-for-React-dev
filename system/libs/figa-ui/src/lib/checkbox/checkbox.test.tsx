import { fireEvent, render, screen } from '@testing-library/react';
import { Checkbox } from './checkbox';

describe('Checkbox can be used when: ', () => {
  it('[FRAGILE] assigns classes', () => {
    const { container, asFragment } = render(<Checkbox className="my-class" />);

    const component = container.querySelector('.checkbox');

    expect(component?.className).toContain('checkbox my-class');
    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] allows to reverse the checkbox label and input', () => {
    const { asFragment } = render(<Checkbox reversed />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] allows to pass custom id', () => {
    const { asFragment } = render(<Checkbox id="my-id" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] skips label rendering if not passed', () => {
    const { asFragment } = render(<Checkbox />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('allows to attach custom event handlers to checkbox input', () => {
    const spy = jest.fn();

    render(<Checkbox label="my-label" onChange={spy} />);

    fireEvent.click(screen.getByText(/my-label/));

    expect(spy).toHaveBeenCalledTimes(1);
  });
});
