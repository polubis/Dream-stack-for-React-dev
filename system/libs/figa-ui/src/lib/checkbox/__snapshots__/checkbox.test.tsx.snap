// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox can be used when:  [FRAGILE] allows to pass custom id 1`] = `
<DocumentFragment>
  <div
    class="checkbox"
  >
    <input
      id="my-id"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;

exports[`Checkbox can be used when:  [FRAGILE] allows to reverse the checkbox label and input 1`] = `
<DocumentFragment>
  <div
    class="checkbox reversed"
  >
    <input
      id=":r1:"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;

exports[`Checkbox can be used when:  [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <div
    class="checkbox my-class"
  >
    <input
      id=":r0:"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;

exports[`Checkbox can be used when:  [FRAGILE] skips label rendering if not passed 1`] = `
<DocumentFragment>
  <div
    class="checkbox"
  >
    <input
      id=":r3:"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
